src/pykeen/nn/functional.py

```diff
@@ -337,22 +335,22 @@ def ermlp_interaction(
     :param final:
         The second linear layer.
 
-    :return: shape: (batch_size, num_heads, num_relations, num_tails)
+    :return: shape: batch_dims
         The scores.
     """
     # same shape
+    prefix = h.shape[:-1]
+    dim = h.shape[-1]
     if h.shape == r.shape and h.shape == t.shape:
-        return final(
-            activation(hidden(torch.cat([h, r, t], dim=-1).view(-1, 3 * h.shape[-1]))),
-        ).view(*h.shape[:-1])
+        return final(activation(hidden(torch.cat([h, r, t], dim=-1).view(-1, 3 * dim)))).view(prefix)
 
-    hidden_dim = hidden.weight.shape[0]
     # split, shape: (embedding_dim, hidden_dim)
-    head_to_hidden, rel_to_hidden, tail_to_hidden = hidden.weight.t().split(h.shape[-1])
-    bias = hidden.bias.view(1, 1, 1, 1, -1)
-    h = h @ head_to_hidden.view(1, 1, 1, h.shape[-1], hidden_dim)
-    r = r @ rel_to_hidden.view(1, 1, 1, r.shape[-1], hidden_dim)
-    t = t @ tail_to_hidden.view(1, 1, 1, t.shape[-1], hidden_dim)
+    head_to_hidden, rel_to_hidden, tail_to_hidden = hidden.weight.t().split(dim)
+    prefix = [1 for _ in prefix]
```

Comment:
I know it might be more verbose, but making a function "ones_like" might make this more readable

Reply:
1ea0d5d